dtm0/ut/drlink.c

```diff
@@ -142,6 +144,11 @@ void m0_dtm0_ut_drlink_simple(void)
 	}
 	m0_free(op);
 	m0_free(fop);
+	for (i = 0; i < DTM0_UT_DRLINK_SIMPLE_POST_NR; ++i)
+		m0_semaphore_down(&dr_sema);
+
+	m0_be_queue_fini(svc->dos_ut_queue);
+	m0_free(svc->dos_ut_queue);
 
 	m0_ut_dtm0_helper_fini(udh);
 	m0_free(udh);
```

Comment:
What do you think about finalising the queue after the service is stopped?

If DTM0 service is stopped then it means there is no ongoing DTM0-related FOMs. It means no one is using `dos_ut_queue`. So that, we can freely finalise it.

Reply:
I think that would be fine.

Reply:
No luck. With your patch, based on dtm/recovery branch, the panic is still there.
```
----- run_ut -t dtm0-ut:drlink-simple -----
START Iteration: 1 out of 1
dtm0-ut
  drlink-simple  motr[955241]:  2a40  FATAL  [lib/assert.c:50:m0_panic]  panic: (m0_list_is_empty(head)) at m0_list_fini() (lib/list.c:38)  [git: 2.0.0-790-4-g40ca11364-dirty] /var/motr/m0ut/m0trace.955241.2022-06-01-21:30:38
Motr panic: (m0_list_is_empty(head)) at m0_list_fini() lib/list.c:38 (errno: 0) (last failed: none) [git: 2.0.0-790-4-g40ca11364-dirty] pid: 955241  /var/motr/m0ut/m0trace.955241.2022-06-01-21:30:38
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(m0_arch_backtrace+0x20)[0x7f0299d54010]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(m0_arch_panic+0xdf)[0x7f0299d541bf]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(m0_panic+0x127)[0x7f0299d421b7]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(+0x39f34c)[0x7f0299d4634c]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(m0_be_queue_fini+0x131)[0x7f0299c8f341]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(+0x359444)[0x7f0299d00444]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(+0x3594ec)[0x7f0299d004ec]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(+0x36e1e3)[0x7f0299d151e3]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(m0_thread_trampoline+0x5a)[0x7f0299d4956a]
/home/520428/work/cortx-motr/motr/.libs/libmotr.so.2(+0x3add7d)[0x7f0299d54d7d]
/lib64/libpthread.so.0(+0x815a)[0x7f029940d15a]
/lib64/libc.so.6(clone+0x43)[0x7f0297d12dd3]
Motr pid 955241 waits for debugger.
```

Reply:
could you please try the queueit() patch (https://github.com/Seagate/cortx-motr/pull/789) with my patch together? I think it could help. I suspect that the root cause here is that the FOM is still running even after the reqh was stopped.

Reply:
With this patch https://github.com/ivan-alekhin/cortx-motr/commit/d15eb5968b9c9a51e6af1c186b6119647285acd8 ,
tests have passed ~3000 iterations.